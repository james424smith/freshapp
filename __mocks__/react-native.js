import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
