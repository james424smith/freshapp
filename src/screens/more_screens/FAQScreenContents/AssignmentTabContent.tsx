import React from "react";
import { View } from "react-native";
import { assignmentContent } from "./faqContent";
import FaqView from "../../../components/FaqView";
type Props = {
  selectedQuestion?: number;
};

const Assignment = (props: Props) => {
  return (
    <View>
      <FaqView
        content={assignmentContent}
        selectedQuestion={props.selectedQuestion}
      />
    </View>
  );
};

export default Assignment;
