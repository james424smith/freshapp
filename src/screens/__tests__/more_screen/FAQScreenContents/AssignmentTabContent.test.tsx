import React from "react";
import AssignmentTabContent from "../../../more_screens/FAQScreenContents/AssignmentTabContent";
import { render, cleanup } from "@testing-library/react-native";

describe("AssignmentTabContent snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render correct ui for AssignmentTabContent", () => {
    const { toJSON } = render(<AssignmentTabContent />);

    expect(toJSON()).toMatchSnapshot();
  });
});
