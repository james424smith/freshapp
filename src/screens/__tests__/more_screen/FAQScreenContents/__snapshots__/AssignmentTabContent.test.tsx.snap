// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssignmentTabContent snapshot test should render correct ui for AssignmentTabContent 1`] = `
<View>
  <RCTScrollView
    data={
      Array [
        Object {
          "answer": "Please contact your local manning agency or crew manager.  They will update your database file and then you will see these changes in the App.",
          "id": 0,
          "opened": false,
          "question": "I cannot see my flights / details of next assignment?",
        },
        Object {
          "answer": "Presently the App only shows these records for your present / current assignment.  Historical records are not shown",
          "id": 1,
          "opened": false,
          "question": "I cannot see previous payroll / working clothes records?",
        },
        Object {
          "answer": "Go to the ‘Assignment screen’ and click on the map icon (top right)",
          "id": 2,
          "opened": false,
          "question": "How do I see the location of the vessel I am joining?",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    getItemLayout={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        style={null}
      >
        <RNGestureHandlerButton
          collapsable={false}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
          testID="faq-button-0"
        >
          <View
            accessible={true}
            style={
              Object {
                "borderBottomColor": "#C5C5C5",
                "borderBottomWidth": 0.3,
                "opacity": 1,
                "paddingVertical": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginLeft": 22.5,
                  "marginRight": 30,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": undefined,
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "width": 600,
                    },
                  ]
                }
              >
                I cannot see my flights / details of next assignment?
              </Text>
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#8E9397",
                            "fontSize": 30,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "fontFamily": "Material Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNGestureHandlerButton>
      </View>
      <View
        style={null}
      >
        <RNGestureHandlerButton
          collapsable={false}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
          testID="faq-button-1"
        >
          <View
            accessible={true}
            style={
              Object {
                "borderBottomColor": "#C5C5C5",
                "borderBottomWidth": 0.3,
                "opacity": 1,
                "paddingVertical": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginLeft": 22.5,
                  "marginRight": 30,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": undefined,
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "width": 600,
                    },
                  ]
                }
              >
                I cannot see previous payroll / working clothes records?
              </Text>
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#8E9397",
                            "fontSize": 30,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "fontFamily": "Material Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNGestureHandlerButton>
      </View>
      <View
        style={null}
      >
        <RNGestureHandlerButton
          collapsable={false}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
          testID="faq-button-2"
        >
          <View
            accessible={true}
            style={
              Object {
                "borderBottomColor": "#C5C5C5",
                "borderBottomWidth": 0.3,
                "opacity": 1,
                "paddingVertical": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginLeft": 22.5,
                  "marginRight": 30,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": undefined,
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "width": 600,
                    },
                  ]
                }
              >
                How do I see the location of the vessel I am joining?
              </Text>
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#8E9397",
                            "fontSize": 30,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "fontFamily": "Material Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNGestureHandlerButton>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
